lib/include/jxl/encode.h

```diff
@@ -405,10 +411,15 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderSetCodestreamLevel(JxlEncoder* enc,
                                                          int level);
 
 /**
- * Sets lossless/lossy mode for the provided options. Default is lossy.
+ * Enables lossless encoding options override. When enabled, overrides all
```

Comment:
nit: I don't think the concept of "override" is the most clear:  "this setting over here is the distance; this setting over there overrides other settings" is more confusing IMHO than scoping the "distance" setting with the lossless: "this is the lossy distance setting when lossless is disabled". Somehow I just don't find this sentence very clear: "Enables lossless encoding options override."

Reply:
I've updated the comment to be more clear.

We could also think about replacing this function with one without argument (that then directly sets all relevant options rather than be a separate state, since that's much easier to explain), but I'd do that in a separate MR since that's not related to the new modular mode option here (updating the comment is related)